File: 2023/Leetcode/algorithms/easy/0013_roman_to_integer/0013_roman_to_integer.go
func romanToInt(s string) int {
    romanDict := map[byte]int{
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000,
    }

    result := 0
    prevValue := 0

    for i := 0; i < len(s); i++ {
        currentValue := romanDict[s[i]]
        result += currentValue

        if prevValue < currentValue {
            result -= 2 * prevValue
        }

        prevValue = currentValue
    }

    return result
}

File: 2023/Leetcode/algorithms/medium/0013_roman_to_integer/0013_roman_to_integer.cs
public class Solution {
    public int RomanToInt(string s) {
        Dictionary<char, int> romanDict = new Dictionary<char, int>() {
            { 'I', 1 },
            { 'V', 5 },
            { 'X', 10 },
            { 'L', 50 },
            { 'C', 100 },
            { 'D', 500 },
            { 'M', 1000 }
        };
        
        int result = 0;
        int prevValue = 0;
        
        foreach (char c in s) {
            int currentValue = romanDict[c];
            result += currentValue;
            
            if (prevValue < currentValue) {
                result -= 2 * prevValue;
            }
            
            prevValue = currentValue;
        }
        
        return result;
    }
}
